### DISCUSSION TITLE
ComputeFireTimesBetween returns Fire Times that are After the Trigger's EndTimeUtc

### DISCUSSION THREAD

**Discussion by ghost** [quartznet/quartznet#1835] [Q&A]
Hi,

We are trying to add validation code to our app to prevent certain Exceptions that Quartz throws. One is "Based on configured schedule, the given trigger '{trigger.Key}' will never fire."

We let users specify an End Date and then use EndAt when building our triggers.

Prior to saving a trigger to the JobStore, I was hoping I could call the method below on the trigger, and thought it would return null or an empty collection if the trigger was setup in such a way where it would never fire:

            var fireTimes = TriggerUtils.ComputeFireTimesBetween((Quartz.Spi.IOperableTrigger)trigger, null, new DateTimeOffset(DateTime.Now), new DateTimeOffset(DateTime.Now.AddDays(8)));

Unfortunately this returns ALL fire times even if they are AFTER the EndTimeUtc for the trigger.

So I also thought this might work:

            var nextFireTime = trigger.GetNextFireTimeUtc();

But that always returns null.

I also tried this, but it always returns false:

            trigger.GetMayFireAgain()

Question: Are there any methods we can call, prior to saving the trigger to the JobStore, to see if it will ever fire?

Our trigger is built like this:

            var trigger = tb
                .WithIdentity(updateSchedule.GetQuartzTriggerName(), TriggerGroup)
                .ForJob(jobDetail.Key)
                .UsingJobData(jobDataMap)
                .WithCronSchedule(cronExpression, x => x
                    .WithMisfireHandlingInstructionDoNothing()
                    .InTimeZone(_timeZone)
                    )
                .StartAt(new DateTimeOffset(updateSchedule.QuartzRunSettings.StartDateTime))
                .EndAt(updateSchedule.GetScheduleEndDate())
                .WithDescription(updateSchedule.GetScheduleRunSettingsDescription())
                .Build();

Thanks, Brent

**Comment by lahma** [comment]
You can try this:

```csharp
((IOperableTrigger) trigger).ComputeFirstFireTimeUtc(calendar: null)
```

If it returns `null`, Quartz will have a problem.
